src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,40 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(popupElement, handleFormSubmit) {
+    super({ popupElement });
+    this._handleFormSubmit = handleFormSubmit;
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+  }
+
+  setLoading(isLoading, submitSave) {
+    if (isLoading) {
+      this._popupForm.querySelector(".modal__button").textContent =
```

Comment:
`this._popupForm.querySelector(".modal__button")`


**Could be improved**
All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` and so on.